torch.rsqrt does not work when all dimensions are 1

### What happened?

The current implementation of `torch.rsqrt` does not work when the shapes of all input tensor are of shape 1. For example, the snippet below fails
x = torch.randn(1,)
out = torch.rsqrt(x)

with 
```
pointers = {'arg0': 0, 'arg1': 17179869184, 'arg2': 34359738368, 'arg3': 51539607552, ...}, op = 'rsqrt'
dimensions = [1]
inputs = [{'device_layout': SpyreTensorLayout(device_size=[1, 64], dim_map =[0, 0], device_dtype=DataFormats.SEN169_FP16), 'host_size': [1], 'lx_addr': None, 'name': 'arg0', ...}]
outputs = [{'device_layout': SpyreTensorLayout(device_size=[1, 64], dim_map =[0, 0], device_dtype=DataFormats.SEN169_FP16), 'host_size': [1], 'lx_addr': None, 'name': 'arg1', ...}]
reduction = False
kwargs = {'name': 'sdsc_fused_rsqrt_0', 'op_info': {'core_division': [[1, 1], [1, 1]], 'n_cores_used': 1}}
tensors = [{'device_layout': SpyreTensorLayout(device_size=[1, 64], dim_map =[0, 0], device_dtype=DataFormats.SEN169_FP16), 'hos..., 64], dim_map =[0, 0], device_dtype=DataFormats.SEN169_FP16), 'host_size': [1], 'lx_addr': None, 'name': 'arg1', ...}]
data_format = <DataFormats.SEN169_FP16: 0>

    def generate_sfp_op(pointers, *, op, dimensions, inputs, outputs, reduction, **kwargs):
        tensors = inputs + outputs
    
        data_format = inputs[0]["device_layout"].device_dtype
    
        d3 = len(dimensions) >= 3
    
        ndim = len(dimensions)
    
        # implement core division on stick dimension
        cores = 1
    
        if "op_info" in kwargs and "core_division" in kwargs["op_info"]:
            # enable work division for non-reduction only for now
            if not reduction:
                split_idx = len(dimensions) * -1 if d3 else 0  # split along stick dim
                cores = kwargs["op_info"]["core_division"][-1][split_idx]
                # FIXME: cores should be the product of list of splits
    
        # TODO: fix constant generation with multiple cores
        if "op_info" in kwargs and "constants" in kwargs["op_info"]:
            cores = 1
    
        if reduction and tensors[-1]["scale"][-1] >= 0:
            op += "nonstick"
    
        # Get operation dim map from the tensor that represents the operation space
        op_dims_tensor = inputs[0] if reduction else outputs[0]
        dl = op_dims_tensor["device_layout"]
        dim_map = dl.dim_map[::-1][1:]
    
        # Downstream code expects dim_indices to be consecutive, so make them so
        # If this is a problem, we can update DimInfos to handle non-consecutive instead
        reindex_map = {v: k for k, v in enumerate(sorted(dim_map))}
        dim_indices = [reindex_map[x] for x in dim_map]
    
        dim_labels = INPUT_DIM_LABELS[: ndim - 1] + OUTPUT_DIM_LABELS[:1]
        dim_splits = [1] * (ndim - 1) + [cores]
    
        core_id_to_wk_slice = {}
        for i in range(cores):
            core_id_to_wk_slice[str(i)] = {
                str(s): i if s == "out" else 0 for s in dim_labels
            }
    
        # Obtain (padded) dimensions of the op from a spyre tensor layout
        padded_op_dimensions = [1] * len(dimensions)
    
        # Un-tile and put in host order
        sizes = dl.device_size[::-1][1:]
    
        for dim in range(ndim):
            sv = op_dims_tensor["scale"][dim]
>           assert sv >= 0, "Scale value should be non-negative for op_dims_tensor"
                   ^^^^^^^
E           torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
E           AssertionError: Scale value should be non-negative for op_dims_tensor
E           
E           Set TORCHDYNAMO_VERBOSE=1 for the internal stack trace (please do this especially if you're reporting a bug to PyTorch). For even more developer context, set TORCH_LOGS="+dynamo"
```

### What did you expect to happen?

The `torch.sqrt` operation should still happen without an error, just like the PyTorch equivalent.

### How can we reproduce it?

For example, when running the torch.rsqrt [test](https://github.com/torch-spyre/torch-spyre/blob/e2287c2d6060a74efb8ebe33256a64bcfd0e8afc/tests/_inductor/test_inductor_ops.py#L75-L86) with `1d1_abs_nz": (cached_randn(1, abs=True) + FP16_EPS,)`

### Any environmental details we need to know?

I tested this on main

### Anything else we need to know?

_No response_

### Relevant log output

```Shell

```

--- a/tests/_inductor/test_inductor_ops.py
+++ b/tests/_inductor/test_inductor_ops.py
@@ -616,6 +616,7 @@ class TestOps(unittest.TestCase, metaclass=ParameterizedTestMeta):
                 "exp": torch.exp,
             },
             "param_sets": {
+                "1d0": {cached_randn((1,), dtype=torch.float16)},
                 "2d0": {cached_randn((1, 3), dtype=torch.float16)},
                 "2d1": {cached_randn((2, 1), dtype=torch.float16)},
                 "3d0": {cached_randn((1, 3, 4), dtype=torch.float16)},

--- a/torch_spyre/_inductor/spyre_kernel.py
+++ b/torch_spyre/_inductor/spyre_kernel.py
@@ -595,6 +595,11 @@ class SpyreKernel(SIMDKernel[CSEVariable]):
         """
         dim_map = map_dims_to_vars(access.layout, access.index)
         var_map = {v: k for k, v in dim_map.items()}
+
+        # Special case: single dimension of size 1 is not elided by inductor
+        if len(op_dimensions) == 1 and op_dimensions[0].numel == 1:
+            return [access.layout.device_layout.dim_map[0]]
+
         return [
             -3
             if (di.var == self.wildcard)

